src/control/server/ctl_smd_rpc.go

```diff
@@ -334,6 +343,33 @@ func addManageRespIDOnFail(log logging.Logger, res *ctlpb.SmdManageResp_Result,
 	}
 }
 
+// Retry dev-replace requests as state propagation may take some time after set-faulty call has
+// been made to manually trigger a faulty device state.
+func replaceDevRetryBusy(ctx context.Context, log logging.Logger, e Engine, req proto.Message) (res *ctlpb.SmdManageResp_Result, err error) {
+	for try := uint(0); try < uint(maxDevReplaceRetries); try++ {
+		res, err = sendManageReq(ctx, &e, drpc.MethodReplaceStorage, req)
+		if err != nil {
```

Comment:
Probably missing something, but I was expecting the following test:
```suggestion
		if err == nil {
```

Reply:
If error in sending, it should be returned to caller, the response status has to to be checked after reply has been received then retry will continue if response status is busy.

Reply:
Thanks for the explanation.

Reply:
So it's intentional that retry only occurs if the response status returned is "Busy"